lib/matplotlib/path.py

```diff
@@ -206,12 +206,13 @@ def _fast_from_codes_and_verts(cls, verts, codes, internals=None):
         return pth
 
     def _update_values(self):
+        self._simplify_threshold = rcParams['path.simplify_threshold']
         self._should_simplify = (
             rcParams['path.simplify'] and
-            (len(self._vertices) >= 128 and
-             (self._codes is None or np.all(self._codes <= Path.LINETO)))
+            len(self._vertices) >= 128 and
+            (self._codes is None or np.all(self._codes <= Path.LINETO)) and
+            self._simplify_threshold > 0
```

Comment:
> this PR will probably benefit from an entry in the docs whatsnew too

Yep, already have a draft in `doc/users/whats_new/path_simplification_updates.rst`